app/models.py

```diff
@@ -218,6 +218,45 @@ def set_fully_registered(self):
             return
         db.session.add(UserJoinedEvent.from_user(self))
 
+    def match(self, level, limit=10):
+        '''
+        Return a list of users that can answer questions this user wants to
+        learn at the specified level.
```

Comment:
Hmm, this says it's returning a list of users, but it seems like it's actually yielding tuples of users and ... um ... a list containing information about the categories/questions the user can answer, sorted by the number of questions in each category?

It kinda makes me wonder if there's an intermediate `MatchedUserResultSet` class or something we can factor out to make the return value a bit easier understand, but at the very least, documenting the return value a bit better would be really helpful!